### DISCUSSION TITLE
Exclude some collections from changes api using ForAllDocuments()

### DISCUSSION THREAD

**Discussion by bwmeyer** [ravendb/ravendb#13085] [General]
I'm subscribing for document changes by using ForAllDocuments().
So, the client will be informed about every document that has changed on server-side.

I have now added some collections which contains documents for logging purpose only. (They aren't relevant for the client at all.)
As these collections are updated frequently, the server will flood the client with document change notifications.

As I haven't seen an option for filtering ForAllDocuments() server-side, I assume the only way to reduce the notifications is to subscribe the relevant collections individually.
As this will result into more than fifty subscriptions per client and database, I'm curios if this is the best approach.

**Comment by ppekrol** [comment]
Hi, underneath this is still 1 connection, so you can execute collection specific changes without an issue.

**Reply by bwmeyer** [reply]
So, multiple subscriptions are sharing the existing WebSocket of the connection?

**Reply by ayende** [reply]
Yes, that is correct. In general, a single `DocumentStore` has a single `Changes()` connection per database used.
All the subscriptions are tunneled through that.

**Reply by bwmeyer** [reply]
When having multiple subscriptions will they be processed in the same order as for ForAllDocument, or will there be some parallelism.
I will assign the same observer to every subscription, which is not locked by now.

**Reply by ppekrol** [reply]
Each of them subscribes to an event. We will receive change notifications from server and fire relevant events to which you are subscribed for. Change notifications from server are processed one by one.
